gadgets/trace_tcpdrop/program.bpf.c

```diff
@@ -174,6 +175,10 @@ static __always_inline int __trace_tcp_drop(void *ctx, struct sock *sk,
 		gadget_socket_lookup(sk, event->netns_id);
 	if (skb_val != NULL) {
 		event->mount_ns_id = skb_val->mntns;
+		// Use the mount namespace of the socket to filter by container
+		if (gadget_should_discard_mntns_id(event->mount_ns_id))
+			goto cleanup;
```

Comment:
nit, check if we are discarding before assigning the value
```suggestion
		// Use the mount namespace of the socket to filter by container
		if (gadget_should_discard_mntns_id(skb_val->mntns))
			goto cleanup;
		event->mount_ns_id = skb_val->mntns;
```